From a88dd621f60a9ae085db12df2496b5e59f804fbf Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Wed, 17 Sep 2014 16:42:09 +0800
Subject: [PATCH 0716/1543] ENGR00331799-4: ASoC: fsl_asrc: add spba clock
 support

commit ecedd4b80b18886861e7d4fa23e5f7c4d014aefe from
git://git.freescale.com/imx/linux-2.6-imx.git

asrc need to enable the spba clock, when sdma is using share peripheral
script. If don't enable it, may cause the read/write wrong data from/to
register.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
---
 sound/soc/fsl/fsl_asrc.c |    8 ++++++++
 sound/soc/fsl/fsl_asrc.h |    1 +
 2 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/sound/soc/fsl/fsl_asrc.c b/sound/soc/fsl/fsl_asrc.c
index 86f45a1..ac3fa93 100644
--- a/sound/soc/fsl/fsl_asrc.c
+++ b/sound/soc/fsl/fsl_asrc.c
@@ -840,6 +840,12 @@ static int fsl_asrc_probe(struct platform_device *pdev)
 		return PTR_ERR(asrc_priv->ipg_clk);
 	}
 
+	asrc_priv->dma_clk = devm_clk_get(&pdev->dev, "dma");
+	if (IS_ERR(asrc_priv->dma_clk)) {
+		dev_err(&pdev->dev, "failed to get dma script clock\n");
+		return PTR_ERR(asrc_priv->dma_clk);
+	}
+
 	for (i = 0; i < ASRC_CLK_MAX_NUM; i++) {
 		sprintf(tmp, "asrck_%x", i);
 		asrc_priv->asrck_clk[i] = devm_clk_get(&pdev->dev, tmp);
@@ -922,6 +928,7 @@ static int fsl_asrc_runtime_resume(struct device *dev)
 
 	clk_prepare_enable(asrc_priv->mem_clk);
 	clk_prepare_enable(asrc_priv->ipg_clk);
+	clk_prepare_enable(asrc_priv->dma_clk);
 	for (i = 0; i < ASRC_CLK_MAX_NUM; i++)
 		clk_prepare_enable(asrc_priv->asrck_clk[i]);
 
@@ -935,6 +942,7 @@ static int fsl_asrc_runtime_suspend(struct device *dev)
 
 	for (i = 0; i < ASRC_CLK_MAX_NUM; i++)
 		clk_disable_unprepare(asrc_priv->asrck_clk[i]);
+	clk_disable_unprepare(asrc_priv->dma_clk);
 	clk_disable_unprepare(asrc_priv->ipg_clk);
 	clk_disable_unprepare(asrc_priv->mem_clk);
 
diff --git a/sound/soc/fsl/fsl_asrc.h b/sound/soc/fsl/fsl_asrc.h
index af9124e..9cb1689 100644
--- a/sound/soc/fsl/fsl_asrc.h
+++ b/sound/soc/fsl/fsl_asrc.h
@@ -350,6 +350,7 @@ struct fsl_asrc {
 	unsigned long paddr;
 	struct clk *mem_clk;
 	struct clk *ipg_clk;
+	struct clk *dma_clk;
 	struct clk *asrck_clk[ASRC_CLK_MAX_NUM];
 	spinlock_t lock;
 
-- 
1.7.5.4

